pom.xml

```diff
@@ -542,6 +548,10 @@
               <argument>run</argument>
               <argument>--pod=${cryostat.itest.podName}</argument>
               <argument>--name=${cryostat.itest.containerName}</argument>
+              <argument>--health-cmd</argument>
+              <argument>curl --fail http://localhost:8181/health/liveness</argument>
+              <argument>--requires</argument>
```

Comment:
In k8s/ocp deployment, we do not (maybe cannot) specify such requirement that grafana and datasource must be up first right? Maybe itest should set it up this way too?

Reply:
In k8s, the three of these containers are placed within one Pod. They should all have health checks on them, which k8s will use to determine the overall health of the Pod. If any of them are not healthy (ex. not yet running), then the Pod will not be healthy, and k8s will refuse to send traffic to it. If we don't wait for the containers to be healthy and just send traffic, like tests requests in, then the requests could fail because we're just testing things too early. I don't think that's necessarily a productive way to run integration tests...

Reply:
Ahh yes make sense then! So the tests are running within the cryostat container?

Reply:
No, the integration tests run as an external process directly on the host machine, and interact with the integration test container "from the outside" over the network. This setup of waiting for all of the containers to be ready emulates how k8s will wait until the Pod is ready before sending any traffic through.

Reply:
Ahh thanks that clears up lots of stuffs!

Then do we still need `<argument>--requires</argument>` when launching `cryostat` containers? Those 3 cryostat,datasource, and grafana should be launched independently right?

```
--requires strings                         Add one or more requirement containers that must be started before this container will star
```

Reply:
I suppose it can be removed. This just guards against accidentally launching the server container without the two other services that it requires. It doesn't do anything in this case since (as far as I can tell) all of the scripts and setup do properly launch all three containers and wait for them to be ready. This at least means that the tests will fail early if for example `repeated-integration-tests.bash` is modified and no longer starts one of the required services before starting the server. In that scenario, this `--requires` flag should fail the test launch early, rather than letting it try to proceed and generating garbage results because the tests are written assuming that all services are present and ready.

Reply:
Ah yes right! Just thinking that we can remove that requirement check to be as close to k8s :))

Reply:
I don't think being close to how k8s lays out containers and dependencies is necessarily the goal, though. In the end k8s and podman are two different container platforms, so there are differences in how we lay out the deployments and get them into the desired state before testing the expected behaviours. It is an "integration" test after all, meant to check the integration between the various components, and so I think it's fair to have a precondition on the tests that the components have actually been deployed completely.

Reply:
Sounds good! Thanks for explaining. Looks good to me!